### ISSUE TITLE
kernel 5.9.0-rc5-next-20200921+ : PM errors, FW warnings

### ISSUE THREAD

**Issue by perexg** [thesofproject/linux#2461]
<pre>
[    8.331228] sof-audio-pci 0000:00:1f.3: hda codecs found, mask 4
[    8.460992] sof-audio-pci 0000:00:1f.3: Firmware info: version 1:5:0-7e6cd
[    8.460994] sof-audio-pci 0000:00:1f.3: Firmware: ABI 3:16:0 Kernel ABI 3:17:0
[    8.461536] sof-audio-pci 0000:00:1f.3: warning: unknown ext header type 3 size 0x1c
[    8.461570] sof-audio-pci 0000:00:1f.3: warning: unknown ext header type 4 size 0x10
[    8.626497] rt1308 sdw:1:25d:1308:0: runtime PM trying to activate child device sdw:1:25d:1308:0 but parent (sdw-master-1) is not active
[    8.626508] rt1308 sdw:2:25d:1308:0: runtime PM trying to activate child device sdw:2:25d:1308:0 but parent (sdw-master-2) is not active
[    8.863706] sof_sdw sof_sdw: snd_soc_register_card failed -517
</pre>

The driver loads and works, but this should be resolved. Thank you.

**Comment by kv2019i** [comment]
@RanderWang @bardliao @plbossart SDW issue in upstream, please check.

**Comment by bardliao** [comment]
The unknown ext header type warning can also be seen on topic/sof-dev branch and should be fixed by https://github.com/thesofproject/linux/pull/2462.
But I don't know how to reproduce the PM issue so far.

**Comment by perexg** [comment]
It seems that it's not 100% reproducible. I've seen the PM errors only once, but it looks like a race somewhere. The hw is Dell Precision 5750.

**Comment by plbossart** [comment]
@perexg never seen this error on card registration. I wonder if it's a false alarm though, this looks like a deferred probe so it could very well be that we need the child device to be initialized but the master is not complete with its own initialization. I also just checked in the rt1308-sdw.c code, we only enable pm_runtime once we detect the device on the bus, which means the master is active.

we have not changed the pm_runtime stuff in months, so wondering if something changed in the pm_runtime in -next.
Do you see this with the SOF topic/sof-dev branch?

**Comment by RanderWang** [comment]
I can't remember we get this issue. It is so rare. I think it was caused by "[    8.863706] sof_sdw sof_sdw: snd_soc_register_card failed -517".

**Comment by mengdonglin** [comment]
Lower the priority to P3 as the reproduce rate of the remaining SDW PM issue is very low.

**Comment by plbossart** [comment]
@perexg I forgot to update this issue but did reproduce it. This may be solved by https://github.com/thesofproject/linux/pull/2530

With the patches in this PR I no longer see any warnings. We've merged the code and will do more testing to confirm it also fixes some of the timeout issues we've seen, but you're welcome to take the patches to see if this works for you :-)

**Comment by mengdonglin** [comment]
Close as the fix has been merged. Please re-open this issue or file a new one if it can be reproduced later.

**Comment by plbossart** [comment]
just to be clear, the fix has NOT been merged upstream, it's in the queue for SoundWire reviews and will be submitted after the SDCA stuff.
